src/solvers/dg_p4est/dg_2d.jl

```diff
@@ -0,0 +1,154 @@
+
+    size_ = (nnodes(dg), nnodes(dg))
+
+    # Use Tuple `node_indices` and `indexfunction` to copy values
+    # from the correct face and in the correct orientation
+    for i in eachnode(dg), v in eachvariable(equations)
+      interfaces.u[1, v, i, interface] = u[v, indexfunction(primary_indices, size_, 1, i),
+                                              indexfunction(primary_indices, size_, 2, i),
+                                              primary_element]
+
+      interfaces.u[2, v, i, interface] = u[v, indexfunction(secondary_indices, size_, 1, i),
+                                              indexfunction(secondary_indices, size_, 2, i),
+                                              secondary_element]
+    end
+  end
+
+  return nothing
+end
+
+
+function calc_interface_flux!(surface_flux_values,
+                              mesh::P4estMesh{2},
+                              equations, dg::DG, cache)
+  @unpack surface_flux = dg
+  @unpack u, element_ids, node_indices = cache.interfaces
+
+  @threaded for interface in eachinterface(dg, cache)
+    # Get neighboring elements
+    primary_element   = element_ids[1, interface]
+    secondary_element = element_ids[2, interface]
+
+    primary_indices   = node_indices[1, interface]
+    secondary_indices = node_indices[2, interface]
+
+    primary_direction   = indices2direction(primary_indices)
+    secondary_direction = indices2direction(secondary_indices)
+
+    size_ = (nnodes(dg), nnodes(dg))
+
+    # Use Tuple `node_indices` and `indexfunction` to access node indices
+    # at the correct face and in the correct orientation to get normal vectors
+    for i in eachnode(dg)
+      u_ll, u_rr = get_surface_node_vars(u, equations, dg, i, interface)
+
+      normal_vector = get_normal_vector(primary_direction, cache,
+                                        indexfunction(primary_indices, size_, 1, i),
+                                        indexfunction(primary_indices, size_, 2, i),
+                                        primary_element)
+
+      flux_ = surface_flux(u_ll, u_rr, normal_vector, equations)
+
+      # Use Tuple `node_indices` and `indexfunction_surface` to copy flux
+      # to left and right element storage in the correct orientation
+      for v in eachvariable(equations)
+        surface_index = indexfunction_surface(primary_indices, size_, 1, i)
+        surface_flux_values[v, surface_index, primary_direction, primary_element] = flux_[v]
+
+        surface_index = indexfunction_surface(secondary_indices, size_, 1, i)
+        surface_flux_values[v, surface_index, secondary_direction, secondary_element] = -flux_[v]
+      end
+    end
+  end
+
+  return nothing
+end
+
+
+# TODO: Taal dimension agnostic
+function calc_boundary_flux!(cache, t, boundary_condition::BoundaryConditionPeriodic,
+                             mesh::P4estMesh{2}, equations, dg::DG)
+  # @assert isempty(eachboundary(dg, cache))
```

Comment:
This method should get an `error("not implemented")` or be removed entirely if it is not needed (yet)